e2e/test/scenarios/native/native.cy.spec.js

```diff
@@ -544,24 +554,64 @@ describe("scenarios > native question > data reference sidebar", () => {
     });
 
     openNativeEditor();
-    cy.icon("reference").click();
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("2 models");
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Native Products Model").click();
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("A model of the Products table"); // description
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Bobby Tables's Personal Collection"); // collection
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("1 column");
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("RENAMED_ID").click();
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("No description");
+    referenceButton().click();
+
+    sidebar().within(() => {
+      cy.findByText("2 models");
+      cy.findByText("Native Products Model").click();
+      cy.findByText("A model of the Products table"); // description
+      cy.findByText("Bobby Tables's Personal Collection"); // collection
+      cy.findByText("1 column");
+      cy.findByText("RENAMED_ID").click();
+      cy.findByText("No description");
+    });
+  });
+
+  describe("metrics", () => {
+    it("should not show metrics when they are not defined on the selected table", () => {
+      openNativeEditor();
+      referenceButton().click();
+      sidebarHeaderTitle().should("have.text", "Sample Database");
+
+      sidebar().within(() => {
+        cy.findByText("ORDERS").click();
+        cy.findByText(/metric/).should("not.exist");
+      });
+    });
+
+    it("should show metrics defined on tables", () => {
+      createQuestion(ORDERS_SCALAR_METRIC);
+
+      openNativeEditor();
+      referenceButton().click();
+      sidebarHeaderTitle().should("have.text", "Sample Database");
+
+      sidebar().within(() => {
+        cy.findByText("ORDERS").click();
+        cy.findByText("1 metric").should("be.visible");
+
+        cy.findByText("Count of orders").should("be.visible").click();
+        cy.findByText("A metric").should("be.visible");
+
+        cy.log("clicking the title should navigate back");
+        cy.findByText("Count of orders").should("be.visible").click();
+      });
+    });
   });
 });
 
+function referenceButton() {
+  return cy.icon("reference");
+}
+
+function sidebarHeaderTitle() {
+  return cy.findByTestId("sidebar-header-title");
+}
+
+function sidebar() {
```

Comment:
afaik we have a helper `sidebar` 

```
export function sidebar() {
  return cy.get("main aside");
}
```

will it work as well?

Reply:
No that one does not work for this sidebar unfortunately. I renamed this one to make the distinction clearer.